cpp/arcticdb/processing/aggregation.cpp

```diff
@@ -302,36 +306,32 @@ namespace
-                    using RawType = typename decltype(col_tag)::DataTypeTag::raw_type;
-                    using MaybeValueType = MaybeValue<RawType, T>;
+            constexpr auto dynamic_schema_data_type = DataType::FLOAT64;
+            using dynamic_schema_tdt = ScalarTagType<DataTypeTag<dynamic_schema_data_type>>;
+            auto col = std::make_shared<Column>(make_scalar_type(dynamic_schema ? dynamic_schema_data_type: data_type_.value()), unique_values, true, false);
+            auto column_data = col->data();
+            col->set_row_data(unique_values - 1);
+            res.add_column(scalar_field(dynamic_schema ? dynamic_schema_data_type : data_type_.value(), output_column_name.value), col);
+            details::visit_type(*data_type_, [&aggregated_, &column_data, unique_values, dynamic_schema] (auto col_tag) {
+                using col_type_info = ScalarTypeInfo<decltype(col_tag)>;
+                using MaybeValueType = MaybeValue<typename col_type_info::RawType, T>;
+                if(dynamic_schema) {
                     auto prev_size = aggregated_.size() / sizeof(MaybeValueType);
                     auto new_size = sizeof(MaybeValueType) * unique_values;
                     aggregated_.resize(new_size);
-                    auto in_ptr =  reinterpret_cast<MaybeValueType*>(aggregated_.data());
+                    auto in_ptr = reinterpret_cast<MaybeValueType *>(aggregated_.data());
                     std::fill(in_ptr + prev_size, in_ptr + unique_values, MaybeValueType{});
-                    auto col = std::make_shared<Column>(make_scalar_type(DataType::FLOAT64), unique_values, true, false);
-                    auto out_ptr = reinterpret_cast<double*>(col->ptr());
-                    for(auto i = 0u; i < unique_values; ++i, ++in_ptr, ++out_ptr) {
-                        *out_ptr = in_ptr->written_ ? static_cast<double>(in_ptr->value_) : std::numeric_limits<double>::quiet_NaN();                }
-
-                    col->set_row_data(unique_values - 1);
-                    res.add_column(scalar_field(DataType::FLOAT64, output_column_name.value), col);
-                });
-            } else {
-                details::visit_type(*data_type_, [&aggregated_, &data_type_, &res, output_column_name, unique_values] (auto col_tag) {
-                    using RawType = typename decltype(col_tag)::DataTypeTag::raw_type;
-                    auto col = std::make_shared<Column>(make_scalar_type(data_type_.value()), unique_values, true, false);
-                    const auto* in_ptr =  reinterpret_cast<const MaybeValue<RawType, T>*>(aggregated_.data());
-                    auto out_ptr = reinterpret_cast<RawType*>(col->ptr());
-                    for(auto i = 0u; i < unique_values; ++i, ++in_ptr, ++out_ptr) {
-                        *out_ptr = in_ptr->value_;
+                    for (auto it = column_data.begin<dynamic_schema_tdt>(); it != column_data.end<dynamic_schema_tdt>(); ++it, ++in_ptr) {
```

Comment:
Don't hugely mind but snake case type is not the normal pattern. Is there a reason why this diverges?

Reply:
Nope, will change